Pod/Classes/XExtensionItem.h

```diff
@@ -96,10 +140,51 @@ typedef UIImage *(^XExtensionItemThumbnailProvidingBlock)(CGSize suggestedSize,
  */
 @property (nonatomic, copy) NSDictionary *userInfo;
 
+/**
+ Add entries from a dictionary-serializable custom object to this paramter object’s `userInfo` dictionary.
+ 
+ @param dictionarySerializable Object whose entries should be added to this paramter object’s `userInfo` dictionary.
+ */
+- (void)addEntriesToUserInfo:(id <XExtensionItemDictionarySerializing>)dictionarySerializable;
```

Comment:
> I think exposing userInfo still makes sense in case the user wants to add purely custom metadata, right?

Yes, until someone does this:

```objc
[itemSource addUserInfoParameters:tumblrParameters];
itemSource.userInfo = @{ @"my-custom-shit": @"whaddup" }; // Blows out the changes from the above line
```

What about just separating these two completely, and only combining them when it becomes time to create the actual extension item?

```objc
[itemSource addCustomParameters:tumblrParameters]; // Stored separately from `userInfo`
itemSource.userInfo = @{ @"my-custom-shit": @"whaddup" };
```